### PR TITLE
Added implementation for several Interval classes

### PR DESCRIPTION
**Pull request by awmarkley**
Interval classes implemented include:

- Year: The interval from the first second of the year (inclusive) to the first second of the next year (exclusive).
- Decade: The interval from the first second of the decade (inclusive) to the first second of the next decade (exclusive).
- Century: An interval from the first second of the century (inclusive) to the first second of the next century (exclusive).
- TwoDigitYear: creates a one year interval from two digits and the century of another interval. Formally: TwoDigitYear([ABCD-EF-GH,...) : Interval, YZ : Integer) = [ABYZ-01-01, (ABYZ+1)-01-01).

### CODE REVIEW

**Comment by bethard** [comment]
Please add a more descriptive title and description for the pull request.

File: src/test/scala/info/bethard/timenorm/formal/TypesTest.scala [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    assert( fromDecade.start === LocalDateTime.of( 1385, 1, 1, 0, 0, 0, 0 ))
+    assert( fromDecade.end === LocalDateTime.of( 1386, 1, 1, 0, 0, 0, 0 )) 
+    
+    val fromCentury = TwoDigitYear( Century( 23 ), 22 )
+    assert( fromCentury.start === LocalDateTime.of( 2322, 1, 1, 0, 0, 0, 0 ))
+    assert( fromCentury.end === LocalDateTime.of( 2323, 1, 1, 0, 0, 0, 0 )) 
+    
+  }
+  
+}
```
Comment (bethard) [inline src/test/scala/info/bethard/timenorm/formal/TypesTest.scala]: Add a newline at the end of the file.

File: src/main/scala/info/bethard/timenorm/formal/Types.scala [hunk 2]
```diff
@@ -26,27 +27,115 @@ case class SimplePeriod(unit: TemporalUnit, n: Number, modifier: Modifier) exten
...
-case class NextPeriod(interval: Interval, period: Period) extends Interval
-case class NextRepeatingInterval(interval: Interval, repeatingInterval: RepeatingInterval) extends Interval
-case class BeforePeriod(interval: Interval, period: Period) extends Interval
-case class BeforeRepeatingInterval(interval: Interval, repeatingInterval: RepeatingInterval) extends Interval
-case class AfterPeriod(interval: Interval, period: Period) extends Interval
-case class AfterRepeatingInterval(interval: Interval, repeatingInterval: RepeatingInterval) extends Interval
-case class Between(startInterval: Interval, endInterval: Interval) extends Interval
-case class Nth(interval: Interval, value: Int, repeatingInterval: RepeatingInterval) extends Interval
-case class IntervalSubIntervalIntersection(interval: Interval, subInterval: RepeatingInterval) extends Interval
+trait Interval extends Temporal {
```
Comment (bethard) [inline src/main/scala/info/bethard/timenorm/formal/Types.scala]: Add a little documentation for the start and end vars. In particular, you'll want to document that start is inclusive and end is exclusive (like you've done in the Intervals below).

**Comment by awmarkley** [comment]
I _think_ that should cover all the changes. Things I have learned: IntelliJ is incredibly helpful in dealing with Git.

**Comment by awmarkley** [comment]
Restored the original .classpath file, I'm not entirely sure why it's listing that commit as "4 days ago."

**Comment by bethard** [comment]
Closed by d63f1e6cb9d0d553ed5fc1c2b3572a4b078125a2

**Pull request closed without merging**